File: internal/domain/report.go
// Package domain contient les structures principales du domaine métier.
package domain

import (
	"time"
)

// Report représente un signalement effectué par un utilisateur sur un contenu.
type Report struct {
	ID               int       `json:"id"`
	UserID           int       `json:"user_id"`
	ReporterUsername string    `json:"reporter_username"`
	ContentType      string    `json:"content_type"` // "post" ou "comment"
	ContentID        int       `json:"content_id"`
	Reason           string    `json:"reason"`
	Status           string    `json:"status"`
	CreatedAt        time.Time `json:"created_at"`
	ProcessedAt      time.Time `json:"updated_at"`

	// Enrichi dynamiquement selon le type
	ContentText     string  `json:"text"`
	ContentMediaURL *string `json:"image_url"`
}

File: onlyflick-app/lib/features/admin/reports/models/report.dart
class Report {
  final int id;
  final int userId;
  final String reporterUsername;
  final String contentType; // "post" ou "comment"
  final int contentId;
  final String reason;
  final String status;
  final DateTime createdAt;
  final String text;
  final String? imageUrl;

  Report({
    required this.id,
    required this.userId,
    required this.reporterUsername,
    required this.contentType,
    required this.contentId,
    required this.reason,
    required this.status,
    required this.createdAt,
    required this.text,
    required this.imageUrl,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'],
      userId: json['user_id'],
      reporterUsername: json['reporter_username'],
      contentType: json['content_type'],
      contentId: json['content_id'],
      reason: json['reason'],
      status: json['status'],
      createdAt: DateTime.parse(json['created_at']),
      text: json['text'],
      imageUrl: json['image_url'],
    );
  }
}
